File: example/example-java/src/main/java/io/holunda/camunda/bpm/data/example/service/OrderRepository.java
package io.holunda.camunda.bpm.data.example.service;

import io.holunda.camunda.bpm.data.example.domain.Order;
import io.holunda.camunda.bpm.data.example.domain.OrderPosition;
import org.springframework.stereotype.Component;

import java.math.BigDecimal;
import java.sql.Date;
import java.time.Instant;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Component
public class OrderRepository {

  private final Map<String, Order> orders = new HashMap<>();

  public OrderRepository() {
    List<OrderPosition> positions = new ArrayList<>();
    positions.add(new OrderPosition("Pencil", BigDecimal.valueOf(1.50), 2L));
    positions.add(new OrderPosition("Pen", BigDecimal.valueOf(2.10), 2L));
    orders.put("1", new Order("1", Date.from(Instant.now()), positions));
  }


  public Order loadOrder(String orderId) {
    return orders.get(orderId);
  }

}

File: example/example-kotlin/src/main/kotlin/service/OrderRepository.kt
package io.holunda.camunda.bpm.data.example.kotlin.service

import io.holunda.camunda.bpm.data.example.kotlin.domain.Order
import io.holunda.camunda.bpm.data.example.kotlin.domain.OrderPosition
import org.springframework.stereotype.Component
import java.math.BigDecimal
import java.sql.Date
import java.time.Instant

@Component
class OrderRepository {

  val orders = mapOf("1" to Order(
    orderId = "1",
    created = Date.from(Instant.now()),
    positions = listOf(
      OrderPosition(title = "Pencil", netCost = BigDecimal.valueOf(1.50), amount = 2),
      OrderPosition(title = "Pen", netCost = BigDecimal.valueOf(2.10), amount = 2)
    )
  ))


  fun loadOrder(orderId: String): Order {
    return orders.getValue(orderId)
  }
}
